synapse/api/auth.py

```diff
@@ -173,6 +173,21 @@ async def get_user_by_req(
                     parent_span.set_tag("device_id", requester.device_id)
                 if requester.app_service is not None:
                     parent_span.set_tag("appservice_id", requester.app_service.id)
+
+                # Tag any headers that we need to extract from the request. This
+                # is useful to specify any headers that a reverse-proxy in front
+                # of Synapse may be sending to correlate and match up something
+                # in that layer to a Synapse trace. ex. when Cloudflare times
+                # out it gives a `cf-ray` header which we can also tag here to
+                # find the trace.
+                for header_name in self.hs.config.tracing.request_headers_to_tag:
+                    headers = request.requestHeaders.getRawHeaders(header_name)
+                    if headers is not None:
+                        parent_span.set_tag(
+                            SynapseTags.REQUEST_HEADER_PREFIX + header_name,
```

Comment:
I'm not really familiar with opentracing tags, is `request_header.cf-ray` an OK thing to do? Or do we need to normalize that to something like `request_header.cf_ray` or whatever?

Reply:
The tag can be whatever. And since it's the name of a header, I think it's probably easiest to follow with the actual header name. Similar to what we do with function arguments but those happen to conform to the Python snake casing already.

These are the OpenTelemetry docs but here are some naming conventions, https://opentelemetry.io/docs/reference/specification/common/attribute-naming/